framework/auth/cas.py

```diff
@@ -244,11 +245,57 @@ def make_response_from_ticket(ticket, service_url):
     client = get_client()
     cas_resp = client.service_validate(ticket, service_furl.url)
     if cas_resp.authenticated:
-        user = User.load(cas_resp.user)
-        # if we successfully authenticate and a verification key is present, invalidate it
-        if user.verification_key:
-            user.verification_key = None
-            user.save()
-        return authenticate(user, access_token=cas_resp.attributes['accessToken'], response=redirect(service_furl.url))
-    # Ticket could not be validated, unauthorized.
+        user, action = get_user_from_cas_resp(cas_resp)
+        # user found and authenticated
+        if user and action == 'authenticate':
+            # if we successfully authenticate and a verification key is present, invalidate it
+            if user.verification_key:
+                user.verification_key = None
+                user.save()
+            return authenticate(
+                user,
+                cas_resp.attributes['accessToken'],
+                redirect(service_furl.url)
+            )
+        # first time login from remote oauth provider, user does not exist or is not linked
+        if not user and action == 'oauth_first_login':
+            from website.util import web_url_for
+            oauth_user = {
+                'fullname': cas_resp.attributes['name'],
+                'provider': cas_resp.attributes['oauthProvider'],
+                'id': cas_resp.attributes['oauthId'],
+                'access_token': cas_resp.attributes['accessToken'],
```

Comment:
as discussed, this access token is returned by CAS during service validate and it is never used.
we will keep it here for now, will remove in the future.